controllers/glanceapi_controller.go

```diff
@@ -809,6 +810,11 @@ func (r *GlanceAPIReconciler) generateServiceConfig(
 		return err
 	}
 
+	databaseAccount, dbSecret, err := mariadbv1.GetAccountAndSecret(ctx, h, instance.Spec.DatabaseAccount, instance.Namespace)
```

Comment:
@zzzeek I'm wondering if a single `EnsureMariaDBAccount` interface is sufficient for both returning the existing information and create if it doesn't exist: if we can improve the implementation and handle the `getOrCreate` logic there will result in limiting the amount of exposed interfaces.
Note I'm not expecting this to be changed, and I'm fine with current approach, I was just wondering if a generic `EnsureMariaDBAccount` is sufficient to efficiently wrap and handle both the `get` and `creation` logic.

Reply:
@zzzeek this comment applies more to PR https://github.com/openstack-k8s-operators/mariadb-operator/pull/184, sorry for writing it here.

Reply:
do you mean that we'd use the `Database` interface for this operation also ?

Reply:
I meant like re-using 

```golang

databaseAccount, dbSecret, err := mariadbv1.EnsureMariaDBAccount(
     ctx, h,
     instance.Spec.DatabaseAccount,
     instance.Namespace, false,
)
```
provided that the function  efficiently support the `get` logic; otherwise I think it's just fine using the current: 

```golang

databaseAccount, dbSecret, err := mariadbv1.GetAccountAndSecret(
    ctx, h,
    instance.Spec.DatabaseAccount,
    instance.Namespace)
```
I see that both have a similar signature and I was wondering if we can just using a single wrapper to get the same data.

Reply:
oh...OK well they do really different things.  one of them actually generates a username and password, the other does not.     The fact that EnsureMariaDBAccount returns the two records was added later when I noticed in glance operator we happened to need the two objects right there anyway, but as I work within keystone and nova operators today, this is not the case; they do the "ensure" up top, throw away the output, then later on they have template-filling functions that use the GetAccountAndSecret function.

I think it's important that the role of "we are creating a new account / secret here because openstack-operators doesn't do this yet" is kept separate from the role of "we need to populate a config template and we need to know what the username/password is".   We can of course put these two behaviors into one single function, and then use a boolean flag `createIfNotExist` to differentiate the two behaviors, I would be OK with that, though personally I like keeping the two names / functions *really* separate.

Reply:
The idea of EnsureMariaDBAccount is that later it will *not* generate anything, openstack-operator will be tasked with doing the generation, and we will not want to change any downstream code.  So the behavior of EnsureMariaDBAccount is expected to transparently change behind the scenes assuming we go forward with moving MariaDBAccount generation up to openstack-operator.

Reply:
ack thanks for explaining the reason behind that choice.